### PR TITLE
ESLint: Fix `jest/expect-expect` violations

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR fixes all of the [`jest/expect-expect`](https://github.com/jest-community/eslint-plugin-jest/blob/b9faa0f27773465c4cad0116acf36c471ae2d6c1/docs/rules/expect-expect.md) violations that should not exist.

## Why?
Ideally, the project should not have any ESLint errors and warnings. We already removed all errors, and this PR cuts many of the existing warnings.

## How?
We're extending the rule when needed. We're also finishing up a few tests that were not quite finished. We're ignoring the rule where that's the intended way to move forward.

We're leaving three violations of the rule intact - those are expected, since these tests are empty and still have to be written. The rest of the warnings are violating the `jest/no-disabled-tests`, so this PR isn't touching them intentionally. 

## Testing Instructions
* Run `npm run lint:js` and verify that the only `jest/expect-expect` violations are the ones in the screenshot below. 
* Verify all tests still pass.

### Testing Instructions for Keyboard
None.

## Screenshots or screencast 
Screenshot 2023-01-17 at 16 05 06

### CODE REVIEW

File: packages/components/src/form-token-field/test/index.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint jest/expect-expect: ["warn", { "assertFunctionNames": ["expect", "expectTokensToBeInTheDocument", "expectTokensNotToBeInTheDocument", "expectVisibleSuggestionsToBe", "expectEscapedProperly"] }] */
```
Comment (tyxla) [inline packages/components/src/form-token-field/test/index.tsx:1]: We're intentionally extending the list of assert functions since we're using some custom ones in this suite.

File: packages/components/src/form-token-field/test/index.tsx [hunk 2]
```diff
@@ -83,6 +85,17 @@ const expectTokensNotToBeInTheDocument = ( tokensText: string[] ) => {
 	);
 };
 
+const expectEscapedProperly = ( tokenHtml: string ) => {
```
Comment (tyxla) [inline packages/components/src/form-token-field/test/index.tsx:88]: This is being extracted since it's used in 2 tests below.

We're also naming it `expect*` to match the assertion function name convention.

File: packages/components/src/form-token-field/test/index.tsx [hunk 3]
```diff
@@ -1233,13 +1246,29 @@ describe( 'FormTokenField', () => {
...
+
+			const input = screen.getByRole( 'combobox' );
+
+			// Typing `slov` will match both `Slovenia` and `Slovakia`.
+			await user.type( input, 'slov' );
+
+			// However, `Slovenia` is already selected.
+			expectVisibleSuggestionsToBe( screen.getByRole( 'listbox' ), [
+				'Slovakia',
+			] );
```
Comment (tyxla) [inline packages/components/src/form-token-field/test/index.tsx:1271]: I'm actually writing this test up since previously it was only rendering and testing nothing in particular.

cc @ciampo for feedback, since he recently worked on refactoring the tests.

File: packages/components/src/form-token-field/test/index.tsx [hunk 4]
```diff
@@ -1516,15 +1545,7 @@ describe( 'FormTokenField', () => {
...
 				'1&amp;2&amp;3&amp;4',
-			].forEach( ( tokenHtml ) => {
-				screen.getByText( ( _, node: Element | null ) => {
-					if ( node === null ) {
-						return false;
-					}
-
-					return node.innerHTML === tokenHtml;
-				} );
-			} );
```
Comment (tyxla) [inline packages/components/src/form-token-field/test/index.tsx:1527]: Being extracted to a separate function above, as mentioned above.

File: packages/components/src/form-token-field/test/index.tsx [hunk 5]
```diff
@@ -1546,15 +1567,7 @@ describe( 'FormTokenField', () => {
...
```
Comment (tyxla) [inline packages/components/src/form-token-field/test/index.tsx:1557]: Being extracted to a separate function above, as mentioned above.

File: packages/components/src/form-token-field/test/index.tsx [hunk 6]
```diff
@@ -1516,15 +1545,7 @@ describe( 'FormTokenField', () => {
...
+			].forEach( ( tokenHtml ) => expectEscapedProperly( tokenHtml ) );
```
Comment (tyxla) [inline packages/components/src/form-token-field/test/index.tsx:1548]: We could shorten this to:

```suggestion
			].forEach( expectEscapedProperly );
```

however this will fail the linter, since it specifically aims to see the invocation of an assertion function.

So I'm going with the spread out version intentionally.

File: packages/components/src/form-token-field/test/index.tsx [hunk 7]
```diff
@@ -1546,15 +1567,7 @@ describe( 'FormTokenField', () => {
...
```
Comment (tyxla) [inline packages/components/src/form-token-field/test/index.tsx:1570]: Just as above.

File: packages/components/src/ui/context/test/context-connect.tsx [hunk 8]
```diff
@@ -10,6 +10,7 @@ import { contextConnect, contextConnectWithoutRef } from '../context-connect';
 import type { WordPressComponentProps } from '../wordpress-component';
 
 // Static TypeScript tests
+/* eslint-disable jest/expect-expect */
```
Comment (tyxla) [inline packages/components/src/ui/context/test/context-connect.tsx:13]: This contains static typescript tests, so intends to use TS for checking for failures, and not Jest. While I have mixed feelings about this manifestation of tests, this intentionally doesn't contain assertion functions, so we're intentionally disabling the rule here.

File: packages/components/src/ui/context/test/wordpress-component.tsx [hunk 9]
```diff
@@ -14,6 +14,7 @@ import { forwardRef } from '@wordpress/element';
 import type { WordPressComponentProps } from '../wordpress-component';
 
 // Static TypeScript checks
+/* eslint-disable jest/expect-expect */
```
Comment (tyxla) [inline packages/components/src/ui/context/test/wordpress-component.tsx:17]: This contains static typescript tests, so intends to use TS for checking for failures, and not Jest. While I have mixed feelings about this manifestation of tests, this intentionally doesn't contain assertion functions, so we're intentionally disabling the rule here.

File: packages/custom-templated-path-webpack-plugin/test/index.js [hunk 10]
```diff
@@ -32,7 +32,9 @@ describe( 'CustomTemplatedPathPlugin', () => {
 	} );
 
 	it( 'should resolve with basename output', async () => {
-		await webpackAsync( config );
-		await accessAsync( outputFile );
+		expect( async () => {
```
Comment (tyxla) [inline packages/custom-templated-path-webpack-plugin/test/index.js:35]: This actually meant to verify that it's not throwing in the meantime, so we're adding that assertion to make it a full-featured test.

File: packages/editor/src/components/editor-help/test/index.native.js [hunk 11]
```diff
@@ -36,11 +36,11 @@ it( 'navigates back from help topic detail screen', async () => {
...
-	await waitForElementToBeRemoved( () =>
-		screen.getByText(
-			'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.'
-		)
-	);
+	const text =
+		'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.';
+	await waitForElementToBeRemoved( () => screen.getByText( text ) );
+
+	expect( screen.queryByText( text ) ).toBeNull();
```
Comment (tyxla) [inline packages/editor/src/components/editor-help/test/index.native.js:43]: I'm adding the assertion here to make it a full-featured test. We could use `.not.toBeInTheDocument()` but that's not supported by our custom React Native implementation yet, thus the `toBeNull()` usage.

File: test/integration/full-content/full-content.test.js [hunk 12]
```diff
@@ -221,76 +221,80 @@ describe( 'full post content fixture', () => {
...
-			throw new Error(
-				'Problem(s) with fixture files:\n\n' + errors.join( '\n' )
-			);
-		}
+			if ( errors.length ) {
+				throw new Error(
+					'Problem(s) with fixture files:\n\n' + errors.join( '\n' )
+				);
+			}
+		} ).not.toThrow();
```
Comment (tyxla) [inline test/integration/full-content/full-content.test.js:298]: All we're doing here is making sure this doesn't throw. That's what the test intended, since it throws itself. 

It looks like a big changeset, but most of it is just whitespace.

**Review by Mamaduka — APPROVED** [review]
LGTM. Thank you, @tyxla!
